packages/nouns-subgraph/src/nouns-dao.ts

```diff
@@ -57,60 +62,95 @@ import {
+  proposal.createdTimestamp = event.block.timestamp;
+  proposal.createdBlock = event.block.number;
+  proposal.lastUpdatedTimestamp = event.block.timestamp;
+  proposal.lastUpdatedBlock = event.block.number;
+  proposal.createdTransactionHash = event.transaction.hash;
+  proposal.startBlock = event.params.startBlock;
+  proposal.endBlock = event.params.endBlock;
+  proposal.forVotes = BIGINT_ZERO;
+  proposal.againstVotes = BIGINT_ZERO;
+  proposal.abstainVotes = BIGINT_ZERO;
+  let desc = event.params.description.split('\\n').join('\n');
+  proposal.description = desc;
+  proposal.title = extractTitle(desc);
+  proposal.status = event.block.number >= proposal.startBlock! ? STATUS_ACTIVE : STATUS_PENDING;
+  proposal.objectionPeriodEndBlock = BIGINT_ZERO;
+
+  const governance = getGovernanceEntity();
+  proposal.totalSupply = governance.totalTokenHolders;
+  const nounsDAO = NounsDAO.bind(event.address);
+  let adjustedSupplyResult = nounsDAO.try_adjustedTotalSupply();
+
+  if (!adjustedSupplyResult.reverted) {
+    proposal.adjustedTotalSupply = adjustedSupplyResult.value;
+  } else {
+    proposal.adjustedTotalSupply = proposal.totalSupply;
+  }
+
+  if (
+    governance.voteSnapshotBlockSwitchProposalId.equals(BIGINT_ZERO) ||
+    BigInt.fromString(proposal.id).lt(governance.voteSnapshotBlockSwitchProposalId)
+  ) {
+    proposal.voteSnapshotBlock = proposal.createdBlock;
+  } else {
+    proposal.voteSnapshotBlock = proposal.startBlock;
+  }
+
+  const dynamicQuorum = getOrCreateDynamicQuorumParams();
+  proposal.minQuorumVotesBPS = dynamicQuorum.minQuorumVotesBPS;
+  proposal.maxQuorumVotesBPS = dynamicQuorum.maxQuorumVotesBPS;
+  proposal.quorumCoefficient = dynamicQuorum.quorumCoefficient;
+
+  proposal.clientId = 0;
+  proposal.save();
+
+  captureProposalVersion(
+    event.transaction.hash.toHexString(),
+    event.logIndex.toString(),
+    proposal,
+    false,
+  );
+}
+
 export function handleProposalCreatedWithRequirements(
   event: ProposalCreatedWithRequirements1,
 ): void {
-  handleProposalCreated(ParsedProposalV3.fromV1Event(event));
+  handleParsedProposal(ParsedProposalV3.fromV1Event(event));
 }
 
 export function handleProposalCreatedWithRequirementsV3(
   event: ProposalCreatedWithRequirements,
 ): void {
-  handleProposalCreated(ParsedProposalV3.fromV3Event(event));
+  handleParsedProposal(ParsedProposalV3.fromV3Event(event));
 }
 
-export function handleProposalCreatedOnTimelockV1(event: ProposalCreatedOnTimelockV1): void {
-  let proposal = getOrCreateProposal(event.params.id.toString());
-  proposal.onTimelockV1 = true;
-  proposal.save();
+export function handleProposalCreatedWithRequirementsV4(
+  event: ProposalCreatedWithRequirementsV4,
+): void {
+  handleParsedProposal(ParsedProposalV3.fromV4Event(event));
 }
 
-export function handleProposalCreated(parsedProposal: ParsedProposalV3): void {
+export function handleParsedProposal(parsedProposal: ParsedProposalV3): void {
```

Comment:
would maybe rename `handleParsedProposal` to something like `saveProposalExtraDetails`

Reply:
done